tests/Nox.Types.Tests/Types/Weight/WeightTests.cs

```diff
@@ -1,11 +1,219 @@
     {
+        void Test()
+        {
+            var weight = Weight.FromKilograms(104.55);
+            weight.ToString().Should().Be("104.55 kg");
+        }
+
+        TestUtility.RunInCulture(Test, culture);
+    }
+
+    [Theory]
+    [InlineData("en-US", "104.55 kg")]
+    [InlineData("pt-PT", "104,55 kg")]
+    public void Weight_ValueInKilograms_ToString_IsCultureDependent(string culture, string expected)
+    {
+        void Test()
+        {
+            var weight = Weight.FromKilograms(104.55);
+            weight.ToString(new CultureInfo(culture)).Should().Be(expected);
+        }
+
+        TestUtility.RunInCulture(Test, culture);
+    }
+
+    [Theory]
+    [InlineData("en-US")]
+    [InlineData("pt-PT")]
+    public void Weight_ValueInPounds_ToString_IsCultureIndependent(string culture)
+    {
+        void Test()
+        {
+            var weight = Weight.FromPounds(230.493295);
+            weight.ToString().Should().Be("230.493295 lb");
+        }
+
+        TestUtility.RunInCulture(Test, culture);
+    }
+
+    [Theory]
+    [InlineData("en-US", "230.493295 lb")]
+    [InlineData("pt-PT", "230,493295 lb")]
+    public void Weight_ValueInPounds_ToString_IsCultureDependent(string culture, string expected)
+    {
+        void Test()
+        {
+            var weight = Weight.FromPounds(230.493295);
+            weight.ToString(new CultureInfo(culture)).Should().Be(expected);
+        }
+
+        TestUtility.RunInCulture(Test, culture);
+    }
+
+    [Fact]
+    public void Weight_Equality_SpecifyingWeightUnit_WithSameUnit_Tests()
+    {
+        var weight1 = Weight.FromKilograms(104.55);
+
+        var weight2 = Weight.FromKilograms(104.55);
+
+        AssertAreEquivalent(weight1, weight2);
+    }
+
+    [Fact]
+    public void Weight_Equality_SpecifyingWeightUnit_WithDifferentUnit_Tests()
+    {
+        var weight1 = Weight.FromKilograms(104.55);
+
+        var weight2 = Weight.FromPounds(230.493295);
+
+        AssertAreEquivalent(weight1, weight2);
+    }
+
+    [Fact]
+    public void Weight_NonEquality_SpecifyingWeightUnit_WithSameUnit_Tests()
+    {
+        var weight1 = Weight.FromKilograms(104.55);
+
+        var weight2 = Weight.FromKilograms(230.493295);
+
+        AssertAreNotEquivalent(weight1, weight2);
+    }
+
+    [Fact]
+    public void Weight_NonEquality_SpecifyingWeightUnit_WithDifferentUnit_Tests()
+    {
+        var weight1 = Weight.FromKilograms(104.55);
+
+        var weight2 = Weight.FromPounds(104.55);
+
+        AssertAreNotEquivalent(weight1, weight2);
+    }
+
+    private static void AssertAreEquivalent(Weight expected, Weight actual)
+    {
+        actual.Should().Be(expected);
+
+        expected.Equals(actual).Should().BeTrue();
+
+        actual.Equals(expected).Should().BeTrue();
+
+        (expected == actual).Should().BeTrue();
+
+        (expected != actual).Should().BeFalse();
+    }
+
+    private static void AssertAreNotEquivalent(Weight expected, Weight actual)
+    {
```

Comment:
You can/should use AssertionScope for this

Reply:
Thanks for the suggestion. Added.